soc/xtensa/espressif_esp32/esp32s3/Kconfig.series

```diff
@@ -12,6 +12,7 @@ config SOC_SERIES_ESP32S3
 	select XIP if !MCUBOOT
 	select HAS_ESPRESSIF_HAL
 	select CPU_HAS_FPU
+	select NEED_POSIX_C_SOURCE
```

Comment:
Probably should be reduced in scope to when the Wifi driver is selected. @sylvioalves - I'm curious about how this is actually being used in the esp32 wifi driver (hal?).